components/script/dom/document.rs

```diff
@@ -997,6 +1002,27 @@ impl<'a> DocumentHelpers<'a> for &'a Document {
             .filter_map(HTMLIFrameElementCast::to_root)
             .find(|node| node.r().subpage_id() == Some(subpage_id))
     }
+
+    /// Find and cache the first title element in the document.
+    fn update_title_element(self) {
+        let title = self.GetDocumentElement().and_then(|root| {
+            if root.r().namespace() == &ns!(SVG) && root.r().local_name() == &atom!("svg") {
+                // Step 1.
+                NodeCast::from_ref(root.r()).child_elements().find(|node| {
+                    node.r().namespace() == &ns!(SVG) &&
+                    node.r().local_name() == &atom!("title")
+                }).map(NodeCast::from_root)
+            } else {
+                // Step 2.
+                NodeCast::from_ref(root.r())
+                         .traverse_preorder()
+                         .find(|node| node.r().is_htmltitleelement())
+            }
+        });
+
+        self.current_title_element.set(
+            title.r().and_then(HTMLTitleElementCast::to_ref).map(JS::from_ref));
```

Comment:
So it turns out this line contains the bug responsible for the two failing subtests in /html/dom/documents/dom-tree-accessors/document.title-09.html I mentioned. In the case of the SVG namespace, the Node returned by searching the `child_elements()` of `root.r()` does not pass the check `is_htmltitleelement()`. This check is used under the hood of `HTMLTitleElementCast::to_ref` when casting a Node to an HTMLTitleElement, such that the result of the cast, when the argument does not pass `is_htmltitleelement()` is `None`. This then breaks the subsequent calls to `Title()` in the test, because we've cached `None` when we should have cached an `HTMLTitleElement`.

@nox I'm not sure what the right solution to this issue is. Should `document.current_title_element` be of type `MutNullableHeap<JS<Node>>`, or `MutNullableHeap<JS<Element>>`, so that we can avoid the explicit cast to HTMLTitleElement? Or maybe `is_htmltitleelement()` should accommodate title elements of the SVG namespace? Or maybe SVG title elements aren't (and shouldn't be) actual HTMLTitleElements?